sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -309,17 +309,11 @@ trait DivModLike extends BinaryArithmetic {
 
   override def nullable: Boolean = true
 
-  final override def eval(input: InternalRow): Any = {
-    val input2 = right.eval(input)
-    if (input2 == null || input2 == 0) {
```

Comment:
ah I see the difference now. Previously we can skip evaluating `input1` if `input2` is 0. Can we change it back and add comment to explain it? sorry for the back and forth!